core/controllers/feedback.py

```diff
@@ -108,3 +114,78 @@ def get(self):
             'cursor': new_urlsafe_start_cursor,
             'more': more,
         })
+
+
+class SuggestionHandler(base.BaseHandler):
+    """"Handles operations relating to learner suggestions."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+
+    @base.require_user
+    def post(self, exploration_id):
+        feedback_services.create_suggestion(
+            exploration_id,
+            self.user_id,
+            self.payload.get('exploration_version'),
+            self.payload.get('state_name'),
+            self.payload.get('suggestion_content'))
+        self.render_json(self.values)
+
+
+class SuggestionActionHandler(base.BaseHandler):
+    """"Handles actions performed on threads with suggestions."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+    ACCEPT_ACTION = 'accept'
+    REJECT_ACTION = 'reject'
+
+    @editor.require_editor
+    def put(self, exploration_id, thread_id):
+        action = self.payload.get('action')
+        if action == self.ACCEPT_ACTION:
+            exp_services.accept_suggestion(
+                self.user_id,
+                thread_id,
+                exploration_id,
+                self.payload.get('commit_message'))
+        elif action == self.REJECT_ACTION:
+             exp_services.reject_suggestion(thread_id, exploration_id)
+        else:
+            raise self.InvalidInputException('Invalid action.')
+
+        self.render_json(self.values)
+
+
+class SuggestionListHandler(base.BaseHandler):
+    """Handles operations relating to list of threads with suggestions."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+    OPEN_LIST_TYPE = 'open'
```

Comment:
_LIST_TYPE_OPEN, etc. (use a common prefix instead of a common suffix, and add a leading underscore)